### PR TITLE
Week 3 Step 6 pt1, site redesign

### PR DESCRIPTION
**Pull request by ethanimooney**
TLDR; Add functionality to limit number of comments displayed, also redesign for site.
LOOK AT LINK FIRST (it's very different): http://ethanimooney-step-2020.appspot.com/

**ListCommentsServlet.java** - Now can take a num  value from the get request and use it and FetchOptions to only retrieve that number of comments.

**NewCommentServlet.java** - changed redirect to new comments.html page.

**comments.html** - New page that now houses the comments feature.

**index.html** - New file, replaces indexOld.html as the new homepage. It looks like it's really weird and large but its fairly simple, I used a ul and li as my main way to display content, with spans to change the color or certain parts of the text. I also used nbsp; to simulate tab spaces

**indexOld.html** - the old index.html file

**script.js** - now passes a number with the /load-comments fetch request, for LoadCommentsServlet to determine how many comments to load. Added two eventListeners as a way to resize the form inputs in comment.html as they are typed in. Changed the way comments look when displayed.

**style.css** - New file, replaces styleOld.css as the new style file.

**styleOld.css** - the old style.css file.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -25,26 +25,73 @@ function changeTextBack(text) {
...
+  commentElement.appendChild(tabs);
+  commentElement.appendChild(tabs);
+  commentElement.appendChild(messageTitle);
+  commentElement.appendChild(spanTwo);
+  spanTwo.appendChild(messageElement);
+  commentElement.appendChild(semi);
+  commentElement.appendChild(br2);
+  commentElement.appendChild(rightBrace);
+  commentElement.appendChild(br3);
+  commentElement.appendChild(br4);
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:95]: Perhaps an array would be more efficient?

File: portfolio/src/main/webapp/indexOld.html [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+<!-- 
+  TODO: (https://github.com/ethanimooney/step-repo/issues/10) Rewrite file to fix the tab space errors in GitHub. Will require entirely new file.
+-->
+
+<!DOCTYPE html>
+<html>
+
+<head>
+	<meta charset="UTF-8">
+	<title>Ethan's Portfolio</title>
```
[the remark this answers is not in the document]
Comment (shaquille-hall) [reply portfolio/src/main/webapp/indexOld.html:46]: If there's not a good reason to keep it, it's ok to just remove. If it's needed later there are ways to recover it.

File: portfolio/src/main/webapp/comments.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            &nbsp;&nbsp;enter-a-message: <span class="input-wrap-message"><span class="extra-width-message" aria-hidden="true"></span><input class="input" type="text" id="comment-form-message" name="comment-form-message"></span>;</br>
+            &nbsp;&nbsp;submit-form:<input type="submit" value="submit">;</br>
+            }</br></br>
+          </li>
+        </form>
+
+        <!--Comment Display-->
+        <li class="text-comment">
+          </br>
+          /* Comments */</br>
```
[the remark this answers is not in the document]
Comment (shaquille-hall) [reply portfolio/src/main/webapp/comments.html:88]: Was this and the comment above addressed?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -25,26 +25,73 @@ function changeTextBack(text) {
...
+elName.addEventListener("keyup", () => {
+  extraWidthName.innerHTML = elName.value;
+});
+
+//Form Text Input Resizing Function - Message
+let elMessage = document.querySelector(".input-wrap-message .input");
+let extraWidthMessage = document.querySelector('.input-wrap-message .extra-width-message');
+elMessage.addEventListener("keyup", () => {
+  extraWidthMessage.innerHTML = elMessage.value;
+});
```
[the remark this answers is not in the document]
Comment (shaquille-hall) [reply portfolio/src/main/webapp/script.js:40]: You can create a wrapper method that adds a layer of abstraction above the event listener. But looking at this now, I don't think that would bring tremendous value here. This is clearer. But if you find you're adding the same listener to multiple (more than 2 or 3) elements that may be useful to avoid code repetition.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -25,26 +25,73 @@ function changeTextBack(text) {
...
+  const commentElement = document.createElement('li');
   commentElement.className = 'comment-element';
 
-  const authorElement = document.createElement('h3');
-  authorElement.innerText = comment.author;
+  const authorElement = document.createTextNode(comment.author);
+
+  const messageElement = document.createTextNode(comment.message);
+
+  const spanOne = document.createElement('span');
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:62]: Please use meaningful variable names. What do these represent?

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (shaquille-hall) [reply portfolio/src/main/webapp/script.js:95]: Yes. Let's chat if you're having difficulties here.

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:62]: They represent the spans for the different colors. I changed them to be more meaningful

**Comment by ethanimooney** [comment]
Latest commit changes the script.js to utilize a different way of displaying comments, updated pom.xml file for the new gcp project, removes indexOld.html (forgot to do styleOld, will do in next commit), and adds a missing piece of styling to style.css

File: portfolio/pom.xml [hunk 6]
```diff
@@ -47,7 +47,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
-          <deploy.projectId>ethanimooney-step-2020</deploy.projectId>
+          <deploy.projectId>ethanimooney-step-apis</deploy.projectId>
```
Comment (shaquille-hall) [inline portfolio/pom.xml:50]: Did this work for you locally? No need to change it if it doesn't bring value.

File: portfolio/src/main/webapp/comments.html [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  </head>
+
+  <body onload="loadComments()">
+    <div class="main-code-body">
+      <ul class="main-code-body-list">
+
+        <!--Navigation-->
+        <li class="text-comment">
+          </br>
+          /* Navigation */</br>
```
[the remark this answers is not in the document]
Comment (shaquille-hall) [reply portfolio/src/main/webapp/comments.html]: Using css class with padding or margins should give the same effect.

File: portfolio/src/main/webapp/styleOld.css [hunk 8]
```diff
@@ -0,0 +1,250 @@
+/* General Style */
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/styleOld.css:1]: I see this also has the "Old" suffix. Is it still being used?

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
+  extraWidthMessage.innerHTML = elMessage.value;
+});
+
 function loadComments() {
-  fetch('/list-comments').then(response => response.json()).then((comments) => {
+  var numComments = document.getElementById("num-comments").value;
+
+  fetch('/list-comments?num=' + numComments).then(response => response.json()).then((comments) => {
     const commentListElement = document.getElementById('comments-list');
+    document.getElementById('comments-list').innerHTML = "";
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:47]: I think you can append the innerHTML call to the line above. If not, you should certainly be able to use the constant you just defined, instead grabbing the raw element again.

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
     comments.forEach((comment) => {
-      commentListElement.appendChild(createCommentElement(comment));
+      const commentElement = document.createElement('p');
+      commentElement.className = 'comment-element';
+      commentElement.appendChild(createDiv(comment.author, 'text-css-label'));
+      commentElement.appendChild(createDiv(comment.message, 'text-css-value'));
+      commentElement.appendChild(createDiv('}', ''));
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:53]: Do you need an opening bracket somewhere at the top of list chain of actions?

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
+      commentListElement.appendChild(commentElement);
     })
   });
 }
 
-function createCommentElement(comment) {
-  const commentElement = document.createElement('div');
-  commentElement.className = 'comment-element';
+function createDiv(passMessage, passClass) {
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:59]: What does "passMessage" (and "passClass") represent? Are you referring to the idea that it's a parameter that's being passed?

File: portfolio/src/main/webapp/script.js [hunk 12]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
+
+  const child = document.createElement('div');
+
+  if (passClass == 'text-css-label') {
+    const labelSpan = document.createElement('span');
+    const bracket = document.createTextNode(" {");
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:65]: Ahh here is the opening bracket. Can you move this to the previous method? It makes it clear what's happening, and avoids comments like the one I left haha.

File: portfolio/src/main/webapp/script.js [hunk 13]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
 
-function createCommentElement(comment) {
-  const commentElement = document.createElement('div');
-  commentElement.className = 'comment-element';
+function createDiv(passMessage, passClass) {
+
+  const child = document.createElement('div');
+
+  if (passClass == 'text-css-label') {
+    const labelSpan = document.createElement('span');
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:64]: Hmm. "labelSpan" doesn't tell me much as a variable name.

File: portfolio/src/main/webapp/script.js [hunk 14]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
+    const bracket = document.createTextNode(" {");
 
-  const authorElement = document.createElement('h3');
-  authorElement.innerText = comment.author;
+    labelSpan.className = passClass;
+    labelSpan.innerText = '.' + passMessage;
+
+    child.appendChild(labelSpan);
+    child.appendChild(bracket);
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:71]: Should the opening bracket be appended first?

File: portfolio/src/main/webapp/script.js [hunk 15]
```diff
@@ -25,26 +25,69 @@ function changeTextBack(text) {
...
+  }
+  else if (passClass == 'text-css-value') {
+    const subChild = document.createElement('span');
+    subChild.className = passClass;
+
+    const title = document.createTextNode('\u00A0\u00A0message: ');
```
Comment (shaquille-hall) [inline portfolio/src/main/webapp/script.js:77]: Please add a comment to identify these values in human readable form.

File: portfolio/src/main/webapp/script.js (hunk 11 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:59]: Yes, the passed message and passed class

File: portfolio/src/main/webapp/styleOld.css (hunk 8 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/styleOld.css:1]: No, forgot to remove it in the last commit, will in the next.

File: portfolio/src/main/webapp/script.js (hunk 15 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:77]: Done

File: portfolio/src/main/webapp/script.js (hunk 14 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:71]: No, since I'm styling it like CSS, the label goes first, then the bracket

File: portfolio/src/main/webapp/script.js (hunk 13 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:64]: Changed

File: portfolio/src/main/webapp/script.js (hunk 12 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:65]: I attempted, since I'm styling this like the CSS, I need the bracket to be on the same line as the name, and the only way to do that is by having it append at the same time.

File: portfolio/src/main/webapp/script.js (hunk 9 above)
Comment (ethanimooney) [reply portfolio/src/main/webapp/script.js:47]: Got it

File: portfolio/pom.xml (hunk 6 above)
Comment (ethanimooney) [reply portfolio/pom.xml:50]: It works, just need billing to be setup

File: portfolio/src/main/webapp/comments.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (ethanimooney) [reply portfolio/src/main/webapp/comments.html]: Alright, changed

File: portfolio/pom.xml (hunk 6 above)
Comment (shaquille-hall) [reply portfolio/pom.xml:50]: Gotcha.

File: portfolio/src/main/webapp/script.js (hunk 10 above)
Comment (shaquille-hall) [reply portfolio/src/main/webapp/script.js:53]: Resolved offline.
